softioc/builder.py

```diff
@@ -24,10 +25,17 @@ def _set_in_defaults(fields):
     fields.setdefault('SCAN', 'I/O Intr')
     fields.setdefault('PINI', 'YES')
     fields.setdefault('DISP', 1)
+    _set_alarm(fields)
 
 def _set_out_defaults(fields):
     fields.setdefault('OMSL', 'supervisory')
 
+def _set_alarm(fields):
+    if "status" in fields:
+        fields['STAT'] = _statStrings[fields.pop('status')]
+    if "severity" in fields:
+        fields['SEVR'] = _severityStrings[fields.pop('severity')]
+
```

Comment:
Might be good to fail at this point of `'STAT'` or `'SEVR'` is already set.  Can't think of an easy way, but the following helper function would do the job:
```python
def _set_field(fields, name, value):
    assert name not in fields, 'Field %s already defined' % name
    fields[name] = value
```

Reply:
I've committed a slight alternative solution, based on similar logic where we already disallow certain keyword pairs. I think this is more in keeping with similar situations, and will provide a clearer error message.